<?xml version="1.0" encoding="utf-8" standalone="yes"?><rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom"><channel><title>try...catch on 老麦的书房</title><link>https://typonotes.com/tags/try...catch/</link><description>Recent content in try...catch on 老麦的书房</description><generator>Hugo -- gohugo.io</generator><language>zh-CN</language><lastBuildDate>Fri, 22 Nov 2024 16:28:41 +0800</lastBuildDate><atom:link href="https://typonotes.com/tags/try...catch/index.xml" rel="self" type="application/rss+xml"/><item><title>Python Keywords: with</title><link>https://typonotes.com/posts/2024/11/22/python-keywords-with/</link><pubDate>Fri, 22 Nov 2024 16:28:41 +0800</pubDate><guid>https://typonotes.com/posts/2024/11/22/python-keywords-with/</guid><description>Python Keywords: with 建议点击 查看原文 查看最新内容。 原文链接: https://typonotes.com/posts/2024/11/22/python-keywords-with/ with 是 python 中的一个关键字。 一种更简单的方式实现 try...catch。 例如， 打开文件后获得文件句柄f， 无论执行是否正常都需要关闭句柄。 1 2 3 4 5 6 try: f = open(&amp;#34;file.txt&amp;#34;, &amp;#34;r&amp;#34;) f.write(&amp;#34;Hello, World!&amp;#34;) finally: f.close() 但是使用 with 关键字， 就可以简单的写成如下 1 2 with open(&amp;#34;file.txt&amp;#34;, &amp;#34;w&amp;#34;) as f: f.write(&amp;#34;Hello, World!&amp;#34;) with...as 的类实现: ContextManager with</description></item></channel></rss>